lite2/auto_client.go

```diff
@@ -0,0 +1,74 @@
+package lite
+
+import (
+	"time"
+
+	"github.com/tendermint/tendermint/types"
+)
+
+// AutoClient can auto update itself by fetching headers every N seconds.
+type AutoClient struct {
+	base         *Client
+	updatePeriod time.Duration
+	quit         chan struct{}
+
+	trustedHeaders chan *types.SignedHeader
+	err            chan error
+}
+
+// NewAutoClient creates a new client and starts a polling goroutine.
+func NewAutoClient(base *Client, updatePeriod time.Duration) *AutoClient {
+	c := &AutoClient{
+		base:           base,
+		updatePeriod:   updatePeriod,
+		quit:           make(chan struct{}),
+		trustedHeaders: make(chan *types.SignedHeader),
```

Comment:
Is that what you do in cosmos-sdk? I always thought plural name is more appropriate because when you iterate over it looks like:

```
for h := trustedHeaders {

}
```